docs/DeveloperGuide.adoc

```diff
@@ -153,6 +163,101 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+
+The following sequence diagram demonstrates the role of the respective components.
+
+image::ViewSequenceDiagram.png[]
+
+==== Design Considerations
+
+===== Aspect: representing the current context state
+
+* **Alternative 1 (current choice):** Creating an immutable `Context` class as a state container to store the appropriate information.
+** Pros: Lightweight; context state can be easily passed between components that require access, as it is encapsulated entirely within `Context`.
+** Cons: Harder to implement.
+
+* **Alternative 2:** Storing the `Context` state in a designated central component, e.g. the `ModelManager`, and exposing relevant methods to other components.
+** Pros: Trivial to implement.
+** Cons: Tightly couples all components that require access to the context state to the central component, complicating testing.
+
+// tag::invitedisinvite[]
+=== [Proposed] Invite/Disinvite feature
+==== Proposed Implementation
+
+The invite/disinvite mechanism is facilitated by `Activity`.
+It extends `AddressBook` with an `ActivityBook`, stored internally as an `activityList`. Additionally, it implements the following operations:
+
+* `Activity#invite()` -- Invites a person to the activity.
+* `Activity#disinvite()` -- Disinvites a person from the activity.
+
+These operations are exposed in the `Activity` class as `Activity#invite()` and `Activity#disinvite()` respectively.
+
+Given below is an example usage scenario and how the invite/disinvite mechanism behaves at each step.
+
+Step 1. The user launches the application for the first time. The `SplitWiser` will be initialized with the initial address book and activity book state.
+
+Step 2. The user executes `activity t/breakfast p/David` command to create an activity named breakfast in the activity book.
+
+Step 3. The user executes `invite p/Louis ...` to invite more people into the current activity. The `invite` command calls `Activity#invite()`, inviting more participants into the activity. The activityList will then be modified which is stored in the ActivityBook.
+
+The following sequence diagram shows how the invite operation works:
+
+image::InviteSequenceDiagram.png[]
+
+NOTE: The lifeline for `InviteCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+
+Step 4. The user decides that inviting the person was a mistake or the particular person has not been involved in any of the expenses in the activity and decides to remove that person by executing the `disinvite` command.
+
+Step 5. The user executes `disinvite p/Louis p/David ...` to remove participant(s) from the current activity. The `disinvite` command calls `Activity#disinvite()`, removing participants from the activity. The activityList will then be modified due to a change in participants which is stored in the ActivityBook.
+
+The following sequence diagram shows how the disinvite operation works:
+
+image::DisinviteSequenceDiagram.png[]
```

Comment:
The two sequence diagrams here are largely similarly. Perhaps you could simply use only 1 of the diagrams, and state that the control flow for the other command is similar.